threads/server-tcp.c

```diff
@@ -0,0 +1,139 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <string.h>
+#include <pthread.h>
+
+/****************************************
+        Author: Tim Wood
+        with a little help from
+        http://beej.us/guide/bgnet/
+****************************************/
+
+#define BACKLOG 10     // how many pending connections queue will hold
+#define MAX_CLIENTS 512
+
+pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
+
+void
+*handle_client(void *arg)
+{
+        pthread_mutex_lock(&lock);
```

Comment:
Hang on, this is from the wrong issue - it bases off of #5. Work is still in progress on that one, I'll take this down and add it again. Sorry